src/MetricExporter.ts

```diff
@@ -225,8 +235,16 @@ export class DynatraceMetricExporter implements MetricExporter {
 
 		function onError(err: Error) {
 			diag.error(err.message);
+
+			if (retries < maxRetries) {
+				// retry after 1000ms.
+				setTimeout(() => process.nextTick(
```

Comment:
why set timeout and next tick? should only need set timeout i think

Reply:
Looks like you are retrying any error. Should we consult with the api team to determine which errors are permanent and not worth retry?

Reply:
Yes, you're right, next tick is not necessary. Removed in 71afcfd.

This only re-tires errors that do not return an error code. Error codes like `401` and `403` which are permanent are already handled in `onResponse()`. I added some tests in 71afcfd to make this a bit clearer.